server/server_test.go

```diff
@@ -1595,6 +1595,35 @@ func (cli *testServerClient) runTestIssue3682(c *C) {
 	c.Assert(err.Error(), Equals, "Error 1045: Access denied for user 'issue3682'@'127.0.0.1' (using password: YES)")
 }
 
+func (cli *testServerClient) runTestIndexScanWithYearCol(c *C) {
+	cli.runTests(c, nil, func(dbt *DBTest) {
+		dbt.mustExec("use test;")
+		dbt.mustExec("drop table if exists t;")
+		dbt.mustExec("create table t (c1 year(4), c2 int, key(c1));")
+		dbt.mustExec("insert into t values(2001, 1);")
+
+		// test inner join
+		rows := dbt.mustQuery("select t1.c1, t2.c1 from t as t1 inner join t as t2 on t1.c1 = t2.c1 where t1.c2 != NULL;")
```

Comment:
We also need to test the explain result.
Please move this test to the package executor. You can take the test func `TestInjectProjBelowTopN` as an example